### PR TITLE
Bump coreutils to 8.27

### PR DESCRIPTION
**Pull request by alarcher**

### CODE REVIEW

File: components/sysutils/coreutils/coreutils.p5m [hunk 1]
```diff
@@ -1,70 +1,34 @@
...
-# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
-# If applicable, add the following below this CDDL HEADER, with the
-# fields enclosed by brackets "[]" replaced with your own identifying
-# information: Portions Copyright [yyyy] [name of copyright owner]
-#
-# CDDL HEADER END
+
 #
-# Copyright (c) 2011, 2013, Oracle and/or its affiliates. All rights reserved.
+# Copyright 2017 <contributor>
```
Comment (pyhalov) [inline components/sysutils/coreutils/coreutils.p5m]: Fix \<contributor\>

File: components/sysutils/coreutils/Makefile [hunk 2]
```diff
@@ -46,16 +51,19 @@ PATCH_LEVEL=		0
 # Prevent gcc register spill error on 32-bit SPARC
 gcc_OPT.sparc.32 =	-O1
 
-CONFIGURE_PREFIX	 =	/usr/gnu
-CONFIGURE_OPTIONS	+=	--libdir=/usr/lib
+# Recreate configure and Makefile to pick up changes in .../src/local.mk
+COMPONENT_PREP_ACTION = (cd $(@D) ; autoreconf -f && automake)
```
Comment (pyhalov) [inline components/sysutils/coreutils/Makefile]: Why should we call automake after autoreconf?

File: components/sysutils/coreutils/Makefile [hunk 3]
```diff
@@ -46,16 +51,19 @@ PATCH_LEVEL=		0
...
 CONFIGURE_OPTIONS	+=	--infodir=$(CONFIGURE_INFODIR)
-CONFIGURE_OPTIONS	+=	CPPFLAGS=-I/usr/include/gmp
 
 # Enable ASLR for this component
 ASLR_MODE = $(ASLR_ENABLE)
 
-COMPONENT_POST_INSTALL_ACTION +=	( cp $(COMPONENT_DIR)/files/*.1  \
-	$(PROTO_DIR)/usr/gnu/share/man/man1/ ) 
+COMPONENT_POST_INSTALL_ACTION += ( cp $(COMPONENT_DIR)/files/*.1 \
+                                      $(PROTOGNUSHAREMAN1) )
```
Comment (pyhalov) [inline components/sysutils/coreutils/Makefile]: Shouldn't we build it 64-bit only?

File: components/sysutils/coreutils/Makefile [hunk 4]
```diff
@@ -46,21 +52,25 @@ PATCH_LEVEL=		0
...
 
-CONFIGURE_PREFIX	 =	/usr/gnu
-CONFIGURE_OPTIONS	+=	--libdir=/usr/lib
+# Recreate configure and Makefile to pick up changes in .../src/local.mk
+COMPONENT_PREP_ACTION = (cd $(@D) ; autoreconf -f && automake)
+
+CONFIGURE_PREFIX	 =	$(GNUDIR)
+CONFIGURE_BINDIR.64	 =	$(GNUBIN)
+CONFIGURE_OPTIONS	+=	CPPFLAGS=" -I$(USRINCDIR)/gmp"
+CONFIGURE_OPTIONS	+=	--libdir=$(USRLIBDIR.$(BITS))
```
Comment (pyhalov) [inline components/sysutils/coreutils/Makefile:61]: Likely, also should set libexecdir.
Also, are there any libstdbuf.so consumers?

Comment (alarcher) [reply components/sysutils/coreutils/Makefile:61]: The path was changed in the manifest in the original recipe. I added the definition of libexecdir now. Do you want the lib in usr/lib/$(MACH64)/coreutils/libstdbuf.so or usr/lib/$(MACH64)/libstdbuf.so?
I do not know other consumers than stdbuf in coreutils, maybe worth asking around.
```
> ldd /usr/lib/* 2> /dev/null | grep libstdbuf
/usr/lib/libstdbuf.so:
> ldd /usr/bin/* 2> /dev/null | grep libstdbuf
>
```

Comment (pyhalov) [reply components/sysutils/coreutils/Makefile:61]: solaris-userland builds both versions and delivers 
```
file $(MACH64)/src/libstdbuf.so path=usr/lib/$(MACH64)/libstdbuf.so
file $(MACH32)/src/libstdbuf.so path=usr/lib/libstdbuf.so
```
Perhaps, it's worth asking.

Comment (alarcher) [reply components/sysutils/coreutils/Makefile:61]: I can do the same if you want.

Comment (alarcher) [reply components/sysutils/coreutils/Makefile:61]: Added 32-bit libstdbuf.

File: components/sysutils/coreutils/Makefile [hunk 5]
```diff
@@ -46,21 +52,27 @@ PATCH_LEVEL=		0
...
 ASLR_MODE = $(ASLR_ENABLE)
 
-COMPONENT_POST_INSTALL_ACTION +=	( cp $(COMPONENT_DIR)/files/*.1  \
-	$(PROTO_DIR)/usr/gnu/share/man/man1/ ) 
+COMPONENT_POST_INSTALL_ACTION += ( cp $(COMPONENT_DIR)/files/*.1 \
+                                      $(PROTOGNUSHAREMAN1) ) 
 
-# common targets
-build:		$(BUILD_32)
+# Build 32-bit for libstdbuf only, then overwrite with 64bit binaries
```
Comment (pyhalov) [inline components/sysutils/coreutils/Makefile]: should we add explicit $(INSTALL_64): $(INSTALL_32) ?

**Comment by alarcher** [comment]
Fixed.

File: components/sysutils/coreutils/Makefile [hunk 6]
```diff
@@ -46,21 +52,30 @@ PATCH_LEVEL=		0
...
+build:		$(BUILD_32_and_64)
 
-# common targets
-build:		$(BUILD_32)
+# Make sure 32-bit binaries are installed first. 
+$(INSTALL_64):	$(INSTALL_32)
 
-install:	$(INSTALL_32)
+install:	$(INSTALL_64)
```
Comment (pyhalov) [inline components/sysutils/coreutils/Makefile]: We build both, but install only 64. And also have rules to install 32-bit version before 64-bit one. This should work, ut seems strange.

Comment (alarcher) [reply components/sysutils/coreutils/Makefile]: OK I misread your message I guess.
Want me to change to install: $(INSTALL_32) $(INSTALL_64)?

Comment (pyhalov) [reply components/sysutils/coreutils/Makefile]: $(INSTALL_32_and_64) as usual

Comment (alarcher) [reply components/sysutils/coreutils/Makefile]: Hmm I thought you suggested earlier that I would change it, that's why I modified.
I will revert then.

Comment (pyhalov) [reply components/sysutils/coreutils/Makefile]: I suggested just to add $(INSTALL_64): $(INSTALL_32) dependency

Comment (alarcher) [reply components/sysutils/coreutils/Makefile]: pyhalov 7 hours ago 
'should we add explicit $(INSTALL_64): $(INSTALL_32) ?'

Comment (pyhalov) [reply components/sysutils/coreutils/Makefile]: And you have it at 76 line

Comment (alarcher) [reply components/sysutils/coreutils/Makefile]: OK :)

File: components/sysutils/coreutils/test/results-i386.master [hunk 7]
```diff
@@ -0,0 +1,905 @@
...
+PASS: tests/tail-2/start-middle.sh
+PASS: tests/touch/60-seconds.sh
+PASS: tests/touch/dangling-symlink.sh
+PASS: tests/touch/dir-1.sh
+PASS: tests/touch/fail-diag.sh
+PASS: tests/touch/fifo.sh
+PASS: tests/touch/no-create-missing.sh
+PASS: tests/touch/no-dereference.sh
+PASS: tests/touch/no-rights.sh
+SKIP: tests/touch/not-owner.sh
```
Comment (pyhalov) [inline components/sysutils/coreutils/test/results-i386.master]: This test passes here.

Comment (alarcher) [reply components/sysutils/coreutils/test/results-i386.master]: How should we deal with this?

Comment (pyhalov) [reply components/sysutils/coreutils/test/results-i386.master]: as a suggestion - grep out only failed tests :)

Comment (alarcher) [reply components/sysutils/coreutils/test/results-i386.master]: Fixed.

**Review by pyhalov — APPROVED** [review]

**Pull request merged**